src/sync-accumulator.ts

```diff
@@ -650,11 +660,14 @@ export class SyncAccumulator {
             Object.keys(roomData._currentState).forEach((evType) => {
                 Object.keys(roomData._currentState[evType]).forEach((stateKey) => {
                     let ev = roomData._currentState[evType][stateKey];
+                    // Push to both fields to provide downgrade compatibility in the sync accumulator db
+                    // the code will prefer `state_after` if it is present
+                    roomJson["org.matrix.msc4222.state_after"]?.events.push(ev);
+                    // Roll the state back to the value at the start of the timeline if it was changed
                     if (rollBackState[evType] && rollBackState[evType][stateKey]) {
-                        // use the reverse clobbered event instead.
                         ev = rollBackState[evType][stateKey];
                     }
-                    roomJson.state.events.push(ev);
+                    roomJson.state?.events.push(ev);
```

Comment:
Why do we need to make this optional? shouldnt we always be able to have a state field?
Either it being a copy of `org.matrix.msc4222.state_after` or the only place for state?